src/openms/source/FORMAT/HANDLERS/MzMLSpectrumDecoder.cpp

```diff
@@ -76,6 +76,20 @@ namespace OpenMS
     }
   }
 
+  inline void fillDataArray(const std::vector<Internal::MzMLHandlerHelper::BinaryData>& data_,
+                            OpenMS::Interfaces::BinaryDataArrayPtr array, bool precision_64, SignedSize index)
+  {
+    // This seems to be the fastest method to move the data (faster than copy or assign)
```

Comment:
I think std::swap would only work if the arrays are of the same type, see http://en.cppreference.com/w/cpp/algorithm/swap so this would work for the 64 bit case but not the 32 bit case I presume

https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/FORMAT/HANDLERS/MzMLHandlerHelper.h#L58 
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/INTERFACES/DataStructures.h#L73